pybamm/solvers/processed_variable.py

```diff
@@ -54,7 +57,27 @@ def __init__(self, base_variable, solution, known_evals=None):
         self.domain = base_variable.domain
         self.auxiliary_domains = base_variable.auxiliary_domains
         self.known_evals = known_evals
-
+        self.warn = warn
+
+        # Set timescale
+        try:
+            self.timescale = solution.model.timescale_eval
+        except AttributeError:
+            if self.warn:
+                pybamm.logger.warning("No timescale provided. Using default of 1 [s]")
+            self.timescale = 1
+
+        # Store spatial variables to get scales
+        self.spatial_vars = {}
+        if solution.model:
+            for var in ["x", "y", "z", "r_n", "r_p"]:
+                if var and var + " [m]" in solution.model.variables:
```

Comment:
this doesn't check that both `var` and `var + " [m]"` are in `solution.model.variables`. It just checks that `var` is not `None` or `False`, and that `var + " [m]"` is in `solution.model.variables`. You have to check both separately.

Reply:
thanks!